velox/common/base/AsyncSource.h

```diff
@@ -153,5 +187,7 @@ class AsyncSource {
   std::function<std::unique_ptr<Item>()> make_;
   std::exception_ptr exception_;
   CpuWallTiming timing_;
+  bool closed_{false};
```

Comment:
Maybe we can add some check in destructor to check it's either moved or closed

Reply:
Added ,thanks.